libr/asm/p/asm_x86_nz.c

```diff
@@ -18,1631 +17,1223 @@ TODO
-		return l;
-	}
-	if (!strcmp (op, "scasd")) {
-		data[l++] = 0xaf;
-		return l;
+static int opadd(RAsm *a, ut8 *data, const Opcode op) {
+	if (op.operands[1].type & OT_CONSTANT) {
+		return process_group_1(a, data, op);
+	} else {
+		return process_1byte_op(a, data, op, 0x00);
 	}
+}
 
-	if (!strcmp (op, "movsb")) {
-		data[l++] = 0xa4;
-		return l;
-	}
-	if (!strcmp (op, "movsw")) {
-		data[0] = 0x66;
-		data[1] = 0xa5;
-		return 2;
-	}
-	if (!strcmp (op, "movsd")) {
-		data[0] = 0xa5;
-		return 1;
-	}
-	if (!strcmp (op, "outsd")) {
-		data[0] = 0x6f;
-		return 1;
-	}
-	if (!strcmp (op, "outsb")) {
-		data[0] = 0x6e;
-		return 1;
-	}
-	if (!strcmp (op, "insb")) {
-		data[0] = 0x6c;
-		return 1;
-	}
-	if (!strcmp (op, "hlt")) {
-		data[0] = 0xf4;
-		return 1;
-	}
-	if (!strcmp (op, "cpuid")) {
-		data[0] = 0xf;
-		data[1] = 0xa2;
-		return 2;
+static int opand(RAsm *a, ut8 *data, const Opcode op) {
+	if (op.operands[1].type & OT_CONSTANT) {
+		return process_group_1(a, data, op);
+	} else {
+		return process_1byte_op(a, data, op, 0x20);
 	}
-	if (!strcmp (op, "fsincos")) {
-		data[0] = 0xd9;
-		data[1] = 0xfb;
-		return 2;
- 	}
-
-	if (!strcmp (str, "call $$")) {
-		memcpy (data, "\xE8\xFF\xFF\xFF\xFF\xC1", 6);
-		return 6;
+}
+
+static int opcmp(RAsm *a, ut8 *data, const Opcode op) {
+	if (op.operands[1].type & OT_CONSTANT) {
+		return process_group_1(a, data, op);
+	} else {
+		return process_1byte_op(a, data, op, 0x38);
 	}
-	if (!strcmp (op, "hang") || !strcmp (str, "jmp $$")) {
-		data[l++] = 0xeb;
-		data[l++] = 0xfe;
-		return l;
+}
+
+static int opsub(RAsm *a, ut8 *data, const Opcode op) {
+	if (op.operands[1].type & OT_CONSTANT) {
+		return process_group_1(a, data, op);
+	} else {
+		return process_1byte_op(a, data, op, 0x28);
 	}
-	if (!strcmp (op, "ud2")) {
-		data[l++] = 0x0f;
-		data[l++] = 0x0b;
-		return l;
+}
+
+static int opor(RAsm *a, ut8 * data, const Opcode op) {
+	if (op.operands[1].type & OT_CONSTANT) {
+		return process_group_1(a, data, op);
+	} else {
+		return process_1byte_op(a, data, op, 0x08);
 	}
-	if (!strncmp (op, "neg ", 4)) {
-		const char *arg = op + 4;
-		int arg0 = getreg (arg);
-		if (a->bits == 64 && *arg == 'r')
-			data[l++] = 0x48;
-		data[l++] = 0xf7;
-		data[l++] = 0xd8 | arg0;
-		return l;
+}
+
+static int opxor(RAsm *a, ut8 * data, const Opcode op) {
+	if (op.operands[1].type & OT_CONSTANT) {
+		return process_group_1(a, data, op);
+	} else {
+		return process_1byte_op(a, data, op, 0x30);
 	}
-	if (!strcmp (op, "rdtsc")) {
-		data[l++] = 0x0f;
-		data[l++] = 0x31;
-		return l;
+}
+
+static int opsbb(RAsm *a, ut8 *data, const Opcode op) {
+	if (op.operands[1].type & OT_CONSTANT) {
+		return process_group_1(a, data, op);
+	} else {
+		return process_1byte_op(a, data, op, 0x18);
 	}
-    if (!strcmp (op, "lfence")) {
+}
+
+static int opbswap(RAsm *a, ut8 *data, const Opcode op) {
+	int l = 0;
+	if (op.operands[0].type | OT_REGALL) {
+		if (op.operands[0].reg == X86R_UNDEFINED) return -1;
```

Comment:
Well there was a time qhere oneliner conditionals where fine without braces. But im starting to get used to this style and i think it makes the code more readable without being too much clunky

> On 05 Jul 2016, at 11:42, Sven Steinbauer <notifications@github.com> wrote:
> 
> In libr/asm/p/asm_x86_nz.c:
> 
> >  	}
> > -    if (!strcmp (op, "lfence")) {
> > +}
> > +
> > +static int opbswap(RAsm *a, ut8 *data, const Opcode op) {
> > +	int l = 0;
> > +	if (op.operands[0].type | OT_REGALL) {
> > +		if (op.operands[0].reg == X86R_UNDEFINED) return -1;
> Alvaro was saying to remove them here
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or mute the thread.
>